### PR TITLE
[vectorized](function) Support array_shuffle and shuffle function.

### PR DESCRIPTION
**Pull request by ZhangYu0123**
# Proposed changes

 Support array_shuffle and shuffle function.

image

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ZhangYu0123** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy [hunk 1]
```diff
@@ -112,6 +112,12 @@ suite("test_array_functions") {
     qt_select "SELECT k1, array_enumerate_uniq(k7) from ${tableName} ORDER BY k1"
     qt_select "SELECT k1, array_enumerate_uniq(k8) from ${tableName} ORDER BY k1"
     qt_select "SELECT k1, array_enumerate_uniq(k10) from ${tableName} ORDER BY k1"
+    qt_select_array_shuffle1 "SELECT k1, k2, array_shuffle(k2, 0), shuffle(k2, 0) from ${tableName} ORDER BY k1"
+    qt_select_array_shuffle2 "SELECT k1, k5, array_shuffle(k5, 0), shuffle(k5, 0) from ${tableName} ORDER BY k1"
```
Comment (zhangstar333) [inline regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:116]: the test case of result maybe add other array function to ensure generate stable results

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by ZhangYu0123** [comment]
run feut

**Comment by ZhangYu0123** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy (hunk 1 above)
Comment (ZhangYu0123) [reply regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy]: done

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by ZhangYu0123** [comment]
run p1

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy (hunk 1 above)
Comment (zhangstar333) [reply regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy]: now I think the implement of array_shuffle is all of ok,
and here I means the function array_shuffle(k2)  in this pr is [3, 2, 1]
Is it possible in other PRs the result is [1,2,3], so can't pass the test?

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by ZhangYu0123** [comment]
run p0

**Review by zhangstar333 — APPROVED** [review]

**Review by Yukang-Lian — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**